common/lib/fdiff.F90

```diff
@@ -12,773 +11,264 @@ module common_finitedifferences
-
- !
- subroutine makeBFDM7P(H2, B, N, alpha, beta, gama, rho, f0, fNp1)
-   implicit none
-   real(dp), intent(inout) :: H2(:,:)
-   real(dp), intent(inout) :: B(:)
-   integer, intent(in) :: N
-   real(dp), intent(in) :: alpha(:)
-   real(dp), intent(in) :: beta(:)
-   real(dp), intent(in) :: gama(:)
-   real(dp), intent(in) :: rho(:)
-   real(dp), intent(in) :: f0
-   real(dp), intent(in) :: fNp1
-
-   integer :: ii
-   real(dp) :: step, step_2, tmp1
-
-   step = 1.0_dp/real(N+1, dp)
-   step_2 = step*step
-
-   !write (*,*) "# 7-point FD, step: ", step
-
-
- ! !========================================================
- ! ! ToDo: do an automatic routine for filling the FD
- ! !       coefficients instead of hard coding (optional)
- ! !========================================================
-
- ! do ii=4, N-3
- ! M(ii,ii) =   alpha(ii)*(-490.0_dp)    + gama(ii)*step_2*180.0_dp
- ! M(ii,ii-1) = alpha(ii)*(270.0_dp)     + beta(ii)*(step)*(-135.0_dp)
- ! M(ii,ii-2) = alpha(ii)*(-27.0_dp)     + beta(ii)*(step)*(27.0_dp)
- ! M(ii,ii-3) = alpha(ii)*(2.0_dp)       + beta(ii)*(step)*(-3.0_dp)
- ! M(ii,ii+1) = alpha(ii)*(270.0_dp)     + beta(ii)*(step)*(135.0_dp)
- ! M(ii,ii+2) = alpha(ii)*(-27.0_dp)     + beta(ii)*(step)*(-27.0_dp)
- ! M(ii,ii+3) = alpha(ii)*(2.0_dp)       + beta(ii)*(step)*(3.0_dp)
- ! end do
-
-!========
-  !  ii=1
-    H2(7, 1) = alpha(1)*(-300.0_dp)     + gama(1)*step_2*180.0_dp + (beta(1)*step*(-150.0_dp))
-    H2(6, 2) = alpha(1)*(90.0_dp)       + beta(1)*step*270.0_dp
-    H2(5, 3) = alpha(1)*(60.0_dp)       + beta(1)*step*(-90.0_dp)
-    H2(4, 4) = alpha(1)*(-15.0_dp)      + beta(1)*step*15.0_dp
-
-  !  ii=2
-    H2(7, 2) = alpha(2)*(-450.0_dp)     + beta(2)*step*(-60.0_dp)
-    H2(6, 3) = alpha(2)*( 240.0_dp)     + beta(2)*step*(180.0_dp)
-    H2(5, 4) = alpha(2)*( -15.0_dp)     + beta(2)*step*(-45.0_dp)
-    H2(4, 5) =                          + beta(2)*step*(6.0_dp)
-    H2(8, 1) = alpha(2)*( 240.0_dp)     + gama(2)*step_2*180.0_dp + (beta(2)*step*(-90.0_dp))
-
-    ii=3
-    H2(7, 3) = alpha(3)*(-490.0_dp)   + gama(3)*step_2*180.0_dp
-    H2(6, 4) = alpha(3)*(270.0_dp)    + beta(3)*(step)*(135.0_dp)
-    H2(5, 5) = alpha(3)*(-27.0_dp)    + beta(3)*(step)*(-27.0_dp)
-    H2(4, 6) = alpha(3)*(2.0_dp)      + beta(3)*(step)*3.0_dp
-    H2(8, 2) = alpha(3)*(270.0_dp)    + beta(3)*(step)*(-135.0_dp)
-    H2(9, 1) = alpha(3)*(-27.0_dp)    + beta(3)*(step)*(27.0_dp)
-
-    tmp1=step_2*180.0_dp
-    B(1) = B(1)*tmp1
-    B(2) = B(2)*tmp1
-    B(3) = B(3)*tmp1
-    B(N-2) = B(N-2)*tmp1
-    B(N-1) = B(N-1)*tmp1
-    B(N) = B(N)*tmp1
+  !> Seven-point finite differences scheme (ref: Bickley).
+  !! On boundaries non-centered five and six point formulae are used.
+  subroutine makeHelmholzFDMatrix7P(M, B, alpha, beta, gama, f0, f1)
+
+    !> finite differences matrix, 7-point scheme
+    real(dp), intent(out), allocatable :: M(:,:)
+
+    !> right-hand side of equation
+    real(dp), intent(inout) :: B(:)
+
+    !>
```

Comment:
Is the purpose of those arguments not known?

Reply:
I tried to understand/comment as much code from the Regensburg-repo as possible, but due to the sheer amount I got sloppy at some point.